### PR TITLE
Fixes test execution 

### PR DESCRIPTION
**Pull request by gurkanindibay**
Tests in tox pipeline does not work. I fixed the problem with this development. Additionally, I performed some optimizations.
Optimizations are documented with in-line comments. Please check "Files Changed" section
Tox test framework deprecated whitelist_externals and started using allowlist_externals as defined below. I fixed this usage
https://github.com/jazzband/django-debug-toolbar/pull/1718

### CODE REVIEW

File: .github/workflows/tox.yml [hunk 1]
```diff
@@ -15,8 +14,8 @@ jobs:
     env:
       PREPARED_STATEMENTS: ${{ matrix.prepared_statements }}
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions/setup-python@v2
+    - uses: actions/checkout@v3
```
Comment (gurkanindibay) [inline .github/workflows/tox.yml:17]: Updated github actions action versions into latest versions

File: .gitignore [hunk 2]
```diff
@@ -4,3 +4,4 @@
 django_multitenant.egg-info/
 .venv
 .tox/
+venv/*
```
Comment (gurkanindibay) [inline .gitignore:7]: Added venv files in ignore list since we will work with venv under project.

File: single-node-docker-compose.yml [hunk 3]
```diff
@@ -6,3 +6,5 @@ services:
     container_name: "${COMPOSE_PROJECT_NAME:-single}_node_postgres"
     ports:
       - "5604:5432"
+    environment:
+      - POSTGRES_HOST_AUTH_METHOD=trust
```
Comment (gurkanindibay) [inline single-node-docker-compose.yml:10]: new citus versions requires password. Without this parameter, db didn't start

File: tox.ini [hunk 4]
```diff
@@ -15,7 +15,7 @@ deps =
 
 setenv =
     PYTHONPATH = {toxinidir}
-whitelist_externals =
+allowlist_externals =
```
Comment (gurkanindibay) [inline tox.ini:18]: parameter name changed for tox. I reflected this parameter change

File: .github/workflows/tox.yml [hunk 5]
```diff
@@ -1,9 +1,8 @@
 on:
   push:
     branches:
-      - master
-  pull_request:
-
+      - "**"
+  workflow_dispatch:
```
Comment (gurkanindibay) [inline .github/workflows/tox.yml:5]: ** provides execution for all branch pushes and workflow_dispatch provides manuel execution

**Review by agedemenli — APPROVED** [review]
LGTM

**Pull request merged**